src/BuiltInTools/dotnet-watch/HotReload/DefaultDeltaApplier.cs

```diff
@@ -44,12 +50,12 @@ public ValueTask InitializeAsync(DotNetWatchContext context, CancellationToken c
                 // Configure the app for EnC
                 context.ProcessSpec.EnvironmentVariables["DOTNET_MODIFIABLE_ASSEMBLIES"] = "debug";
                 context.ProcessSpec.EnvironmentVariables["DOTNET_HOTRELOAD_NAMEDPIPE_NAME"] = _namedPipeName;
+            }
 
-                // If there's any .razor file, we'll assume this is a blazor app and not cause a browser refresh.
-                if (!SuppressBrowserRefreshAfterApply)
-                {
-                    _refreshBrowserAfterFileChange = !context.FileSet.Any(f => f.FilePath.EndsWith(".razor", StringComparison.Ordinal));
-                }
+            // If there's any .razor file, we'll assume this is a blazor app and not cause a browser refresh.
+            if (!SuppressBrowserRefreshAfterApply)
+            {
+                RefreshBrowserAfterFileChange = !context.FileSet.Any(f => f.FilePath.EndsWith(".razor", StringComparison.Ordinal));
```

Comment:
What happens in a Blazor Server app that has a mix of .cshtml and .razor files? We may need to make this decision per edit depending on what files changed.

Reply:
Updated it - we tell the browser and it decides to reload based on the presence of Blazor.